apps/admin_audit/lib/AppInfo/Application.php

```diff
@@ -79,6 +80,10 @@ public function __construct() {
 	}
 
 	public function register(IRegistrationContext $context): void {
+		$context->registerService(LoggerInterface::class, function (ContainerInterface $c) {
```

Comment:
I dislike overload default services.

What I did in Activity with the IDBConnection was specifying my actual implementations, like this:
```suggestion
		$context->registerService(Action::class, function (ContainerInterface $c) {
			return new Action($c->get('AuditLogger'))
		}

		$context->registerService('AuditLogger', function (ContainerInterface $c) {
```

But works better in activity app as it was 2 classes to inject into. Here the action class has 10 classes extending it.

Reply:
The current implementation e.g. breaks when you try to log something to the normal log in the background job etc.

Reply:
Just pushed a new implementation that instead uses a new interface `IAuditLogger`.This has the advantage of enforcing the type of logger used based on the context and moving the logger creation method instead in a new file.